apps/src/sites/studio/pages/foorm/forms/editorHelpers.js

```diff
@@ -20,8 +20,9 @@ const nameKeyValidator = new RegExp(/^[a-z0-9_]+$/i);
 // (non-basic characters can get stripped when being transferred to the server)
 export const lintFoormKeys = (text, options, cm) => {
   const annotations = [];
-  const nameEntries = Array.from(text.matchAll(nameKeyRegex));
-  nameEntries.forEach(match => {
+
+  let match;
+  while ((match = nameKeyRegex.exec(text)) !== null) {
```

Comment:
Why a while loop here?

Reply:
exec stores some state about where it last found a match, and each time you call it, it finds the next match in the string. when it doesn't find any more matches, it'll return null, and we break out of the loop

Reply:
Ooh got it, and I see a similar example in the docs. Thanks!